fcrepo-kernel-impl/src/main/java/org/fcrepo/kernel/impl/FedoraResourceImpl.java

```diff
@@ -465,10 +467,23 @@ public Boolean isNew() {
     public void replaceProperties(final IdentifierConverter<Resource, FedoraResource> idTranslator,
         final Model inputModel, final RdfStream originalTriples) throws MalformedRdfException {
 
+        LOGGER.info("Resource path: {}", getPath());
+
+        final StmtIterator stmtIterator = inputModel.listStatements();
+
+        while (stmtIterator.hasNext()) {
+            final Statement s = stmtIterator.next();
+            final Resource subject = s.getSubject();
+            if (!subject.toString().contains(getPath())) {
```

Comment:
This test is badly wrong.

- It is comparing a JCR value (the result of `getPath()`) with an RDF value (`subject.toString()`). That kind of comparison _must_ be mediated by an `IdentifierConverter`. Use something more like `equals(idTranslator.convert(subject))`, catching exceptions, or pass in the request URI and do a comparison to that.
- It does not account for blank nodes.
- It does not account for hash-URIs